### PR TITLE
docs: glossary, Groups guide

### PR DESCRIPTION
**Pull request by 0xorbus**
- Add incremental Merkle tree.
- Fix case
- Revise off-chain groups section.

### CODE REVIEW

**Comment by 0xorbus** [comment]
@cedoor Please review

File: versioned_docs/version-V2/guides/groups.md [hunk 1]
```diff
@@ -48,29 +61,55 @@ import { Group } from "@semaphore-protocol/group"
 const group = new Group()
 ```
 
-Members can be added one at a time or in batches.
+The following example code passes _`treeDepth`_ to create a group for thirty members:
```
Comment (cedoor) [inline versioned_docs/version-V2/guides/groups.md:64]: It creates a group for 2^30=1073741824 members

**Comment by 0xorbus** [comment]
> 

Can you merge this if it's OK? I can't assign reviewers or merge PRs.

**Comment by cedoor** [comment]
Hey, I left you some review comments. You should be able to read them. Lmk!

**Pull request merged**